lib/wanda/catalog.ex

```diff
@@ -22,26 +24,44 @@ defmodule Wanda.Catalog do
     |> Path.join("/*")
     |> Path.wildcard()
     |> Enum.map(&Path.basename(&1, ".yaml"))
-    |> Enum.map(&get_check(&1))
+    |> get_checks()
   end
 
   @doc """
   Get a check from the catalog.
   """
-  @spec get_check(String.t()) :: Check.t()
+  @spec get_check(String.t()) :: {:ok, Check.t()} | {:error, any}
   def get_check(check_id) do
-    get_catalog_path()
-    |> Path.join("#{check_id}.yaml")
-    |> YamlElixir.read_from_file!()
-    |> map_check()
+    with path <- Path.join(get_catalog_path(), "#{check_id}.yaml"),
+         {:ok, file_content} <- YamlElixir.read_from_file(path),
+         {:ok, check} <-
+           map_check(file_content) do
+      {:ok, check}
+    else
+      {:error, :malformed_check} = error ->
+        Logger.error(
+          "Check with ID #{check_id} is malformed. Check if all the required fields are present."
+        )
+
+        error
+
```

Comment:
we could be more precise about and bubble up a specific error when a check is not found.
By matching:

```elixir 
{:error,
 %YamlElixir.FileNotFoundError{ }}
```
you could return `{:error, :not_found}`

Reply:
on second thoughts, since the `get_checks` function is best-effort, couldn't we just return `nil` here if no check is found? @arbulu89 thoughts?

Reply:
What would be `something goes really wrong`? Can be have something more critical that the check doesn't exist hehe
I don't think this matches that match on Ecto's `get` function, as the `Repo.get` returning `nil` is a normal scenario (you don't match any row with the provided clause), but here we are in an exception. That's why I prefer the `ok, error` option.
Anyway, `get_check` is being used almost as a private function, so we could use either ways.

About being more precise in the error capturing and bubbling up, the same as before. We are using this function almost as a private, so nobody is using the error response for anything more than knowing that it is an error. But well, we can improve it of course for sake of completeness

Reply:
I'm with @arbulu89 , I think we need something specific to match in order to distinguish the "real" error

Reply:
As a comment, I had a discussion with @fabriziosestito , and he explained me some differences about recoverable and non recoverable errors, so we could actually make this function better.
But let's postpone it for future improvements.